### PR TITLE
[Merged by Bors] - chore: remove unnecessary @[eqns] attributes

### LINKED ISSUE #2243: an equation lemma with `autoParam` arguments fails to `rewrite`
### Description

When we use an equation lemma of a definition with `autoParam` arguments in `rewrite` tactic, it fails.

### Steps to Reproduce

```lean
import Lean

open Lean Elab Tactic in
elab "exact_false" : tactic =>
  closeMainGoal (mkConst ``Bool.false)

def f (b : Bool := by exact_false) : Nat := bif b then 1 else 0

example : f false = bif false then 1 else 0 := by rw [f]
example : f true = bif true then 1 else 0 := by rw [f]
example (b : Bool) : f b = bif b then 1 else 0 := by rw [f]
```

**Expected behavior:**

Above all 3 examples succeed.

**Actual behavior:**

```lean
example : f false = bif false then 1 else 0 := by rw [f]
-- failed to rewrite using equation theorems for 'f'
example : f true = bif true then 1 else 0 := by rw [f]
-- failed to rewrite using equation theorems for 'f'
example (b : Bool) : f b = bif b then 1 else 0 := by rw [f]
-- failed to rewrite using equation theorems for 'f'
```

### Versions

Lean (version 4.0.0-nightly-2023-05-16, commit ebc32af2e606, Release)
Windows 11

### PR DESCRIPTION
**Pull request by kim-em**
These attributes are unused in Mathlib.

Many of them were workarounds for the now-resolved leanprover/lean4#2243; this also allows the lemmas themselves (`hasFiniteIntegral_def`, `integrable_def`, `memℒp_def`, and `integrableOn_def`) to be deleted.

We are currently experiencing problems with the `@[eqns]` attribute on the Lean nightlies. I'm uncertain yet what the outcome is going to be there, but it seems prudent to reduce our unnecessary exposure to a language feature added in Mathlib.

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/Notation.lean [hunk 1]
```diff
@@ -135,8 +135,6 @@ theorem ofNat_zero : ofNat 0 = 0 :=
 theorem ofNat_succ (n) : ofNat (Nat.succ n) = oadd 0 n.succPNat 0 :=
   rfl
 
-attribute [eqns ofNat_zero ofNat_succ] ofNat
-
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Ordinal/Notation.lean:139]: There is a porting note on line 130 that refers to this line. I think we should have a tracking issue for "things that `eqns` tried to solve", and update all the existing porting notes near these deleted lines to refer to it.

Comment (eric-wieser) [reply Mathlib/SetTheory/Ordinal/Notation.lean:139]: (https://github.com/leanprover/lean4/issues/2042 was originally intended to be that issue, but I didn't capture the intent quite well enough, and the solution fixed a different space of problems to the one I was trying to report)

Comment (semorrison) [reply Mathlib/SetTheory/Ordinal/Notation.lean:139]: In fact, this one is used, so I'll just revert.

File: Mathlib/MeasureTheory/Function/LpSeminorm/Basic.lean [hunk 2]
```diff
@@ -114,8 +114,6 @@ theorem memℒp_def {α} {_ : MeasurableSpace α} (f : α → E) (p : ℝ≥0∞
     Memℒp f p μ ↔ (AEStronglyMeasurable f μ ∧ snorm f p μ < ∞) :=
   Iff.rfl
 
-attribute [eqns memℒp_def] Memℒp
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Function/LpSeminorm/Basic.lean:117]: The porting note above (leanprover/lean4#2243) suggests that maybe the `_def` lemma itself can be deleted?

File: Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean [hunk 3]
```diff
@@ -185,13 +185,11 @@ theorem rnDeriv_def (s : SignedMeasure α) (μ : Measure α) : rnDeriv s μ = fu
       (s.toJordanDecomposition.negPart.rnDeriv μ x).toReal :=
   rfl
 
-attribute [eqns rnDeriv_def] rnDeriv
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean:188]: The porting note above probably should be updated with a sentence about avoiding `@[eqns]`

File: Mathlib/Computability/TMToPartrec.lean [hunk 4]
```diff
@@ -1251,7 +1251,6 @@ theorem K'.elim_aux (a b c d) : K'.elim a b c d K'.aux = c := rfl
 
 theorem K'.elim_stack (a b c d) : K'.elim a b c d K'.stack = d := rfl
 
-attribute [eqns K'.elim_main K'.elim_rev K'.elim_aux K'.elim_stack] K'.elim
```
Comment (eric-wieser) [inline Mathlib/Computability/TMToPartrec.lean:1254]: The porting note above claims that the equations are being replaced, so is no longer accurate.

File: Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean [hunk 5]
```diff
@@ -178,20 +178,16 @@ def rnDeriv (s : SignedMeasure α) (μ : Measure α) : α → ℝ := fun x =>
     (s.toJordanDecomposition.negPart.rnDeriv μ x).toReal
 #align measure_theory.signed_measure.rn_deriv MeasureTheory.SignedMeasure.rnDeriv
 
--- Porting note: The generated equation theorem is the form of `rnDeriv s μ x`.
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean:181]: Isn't this still true?

File: Mathlib/MeasureTheory/Integral/IntegrableOn.lean [hunk 6]
```diff
@@ -96,8 +96,6 @@ theorem integrableOn_def (f : α → E) (s : Set α) (μ : Measure α) :
     IntegrableOn f s μ ↔ Integrable f (μ.restrict s) :=
   Iff.rfl
 
-attribute [eqns integrableOn_def] IntegrableOn
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Integral/IntegrableOn.lean:99]: `integrableOn_def` can be deleted too

File: Mathlib/MeasureTheory/Function/L1Space.lean [hunk 7]
```diff
@@ -446,8 +444,6 @@ theorem integrable_def {α} {_ : MeasurableSpace α} (f : α → β) (μ : Measu
     Integrable f μ ↔ (AEStronglyMeasurable f μ ∧ HasFiniteIntegral f μ) :=
   Iff.rfl
 
-attribute [eqns integrable_def] Integrable
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Function/L1Space.lean:449]: This one doesn't have a porting note, but I strongly suspect it was another case of leanprover/lean4#2243 and so the lemma can be deleted too.

File: Mathlib/MeasureTheory/Function/L1Space.lean [hunk 8]
```diff
@@ -112,8 +112,6 @@ theorem hasFiniteIntegral_def {_ : MeasurableSpace α} (f : α → β) (μ : Mea
     HasFiniteIntegral f μ ↔ ((∫⁻ a, ‖f a‖₊ ∂μ) < ∞) :=
   Iff.rfl
 
-attribute [eqns hasFiniteIntegral_def] HasFiniteIntegral
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Function/L1Space.lean:115]: idem

File: Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean (hunk 5 above)
Comment (semorrison) [reply Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean:181]: Yes, but I don't see any need to preserve this information.

File: Mathlib/Computability/TMToPartrec.lean [hunk 9]
```diff
@@ -1240,9 +1240,6 @@ def K'.elim (a b c d : List Γ') : K' → List Γ'
   | K'.stack => d
 #align turing.partrec_to_TM2.K'.elim Turing.PartrecToTM2.K'.elim
 
-/- Porting note: The equation lemma of `elim` simplifies to `match` structures. To prevent this,
```
Comment (eric-wieser) [inline Mathlib/Computability/TMToPartrec.lean:1243]: The first sentence of this porting note is still true, so I don't think we should remove it.

File: Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean (hunk 5 above)
Comment (eric-wieser) [reply Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean:181]: this explains why we need the `_def` version at all, and also flags a perceived usability shortcoming in the default equation lemmas.

File: Mathlib/MeasureTheory/Function/L1Space.lean (hunk 8 above)
Comment (semorrison) [reply Mathlib/MeasureTheory/Function/L1Space.lean:115]: This one is used below. If you'd like to remove it, could you please do it yourself?

File: Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean (hunk 5 above)
Comment (semorrison) [reply Mathlib/MeasureTheory/Decomposition/SignedLebesgue.lean:181]: Okay, I restored it as a plain comment.

File: Mathlib/Computability/TMToPartrec.lean (hunk 9 above)
Comment (eric-wieser) [reply Mathlib/Computability/TMToPartrec.lean:1243]: Given we have two other porting notes in this file about equation lemmas, I don't think we should remove this one.

Comment (semorrison) [reply Mathlib/Computability/TMToPartrec.lean:1243]: 🤷 

I've restored it, no longer as a porting note.

**Comment by kmill** [comment]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**